src/components/PrimaryBadgeUpdater.tsx

```diff
@@ -0,0 +1,28 @@
+/*
+ * SPDX-FileCopyrightText: 2023 Zextras <https://www.zextras.com>
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import { useEffect, useMemo } from 'react';
+
+import { useReactiveVar } from '@apollo/client';
+import { updatePrimaryBadge } from '@zextras/carbonio-shell-ui';
+import { find } from 'lodash';
+
+import { uploadVar } from '../carbonio-files-ui-common/apollo/uploadVar';
+import { FILES_ROUTE } from '../carbonio-files-ui-common/constants';
+import { UploadStatus } from '../carbonio-files-ui-common/types/graphql/client-types';
+
+export const PrimaryBadgeUpdater = (): null => {
+	const uploadStatus = useReactiveVar(uploadVar);
+	const isUploadFailed = useMemo(
+		() => find(uploadStatus, (item) => item.status === UploadStatus.FAILED) !== undefined,
```

Comment:
This could be a _some_ instead of a _find_, and we could also remove lodash by using Object.values(uploadStatus)